upstairs/src/lib.rs

```diff
@@ -2544,8 +2487,29 @@ impl Upstairs {
             DsState::WaitQuorum => {
                 assert_eq!(old_state, DsState::WaitActive);
             }
+            DsState::Repair => {
+                assert!(!self.is_active());
+                assert_eq!(old_state, DsState::WaitQuorum);
+            }
             DsState::Replay => {
                 assert_eq!(old_state, DsState::Offline);
+                assert!(self.is_active());
+            }
+            DsState::Active => {
+                if old_state != DsState::WaitQuorum
+                    && old_state != DsState::Repair
+                {
+                    panic!(
+                        "[{}] {} Invalid transition: {:?} -> {:?}",
+                        client_id, self.uuid, old_state, new_state
+                    );
+                }
+                /*
+                 * Make sure repair happens when offline
```

Comment:
I think this should read "Make sure repair happens when inactive", to avoid confusion with the Offline state.

Reply:
Oh yeah, that is a much needed suggestion.

Reply:
Fixed in 2b8628c